packages/integration-sdk-core/src/data/converters.ts

```diff
@@ -179,11 +179,40 @@ export function getTime(
  * Date(time).getTime()`.
  *
  * @param time a time value
+ * @param precision - the precision of the numeric value being supplied. Used to determine how to convert to ms.
+ *          ms - milliseconds
+ *          sec - seconds
  */
 export function parseTimePropertyValue(
   time: Date | string | number | undefined | null,
+  precision?: 'ms' | 'sec',
 ): number | undefined {
   if (time) {
+    // Required because isNaN(new Date()) === false
+    if (time instanceof Date) {
+      return time.getTime();
+    }
+
+    // Convert numeric strings to a float
+    if (typeof time === 'string' && !isNaN(Number(time))) {
+      time = parseFloat(time);
+    }
+
+    // Require precision if time is a number
+    if (typeof time === 'number' && !isNaN(time)) {
+      if (!precision) {
+        throw new Error(
+          'Argument precision is required when parsing a number.',
+        );
+      } else if (precision === 'sec') {
+        time = Number(time) * 1000;
+      } else if (precision === 'ms' && !Number.isInteger(time)) {
+        throw new Error(
```

Comment:
Strictly speaking I believe the Date() constructor will allow a floating point number to be passed to it, it'll just ignore everything after the decimal. Feels more like a time to warn instead of throwing an error.

Reply:
I agree.
I had a failing test case that I was dealing with but this isn't the solution. Correcting